### ISSUE TITLE
Extend test for clustering algorithms

### ISSUE THREAD

**Issue by snath-xoc** [snath-xoc/sample-weight-audit-nondet#10]
So far we only look at KMeans clustering as a "transformer" however could be good to extend tests for clustering algorithms altogether (e.g., DBSCAN)

**Comment by ogrisel** [comment]
Maybe we could do something such as:

- use the classification dataset;
- compute a supervised clustering metric such as the Rand index between `np.repeat(y_test, sample_weight_test)`, `np.repeat(clusterer_weighed/repeated.predict(X_test), sample_weight_test)` on the large (non-subsetted) test set (remark: the `np.repeat` is used because the Rand index computing function [`sklearn.metrics.rand_score`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.rand_score.html#sklearn.metrics.rand_score) does not accept a `sample_weight` parameter).
- collect Rand index values for both `clusterer_weighted` and `clusterer_repeated` with `n_stochastic_fits` random states to collect the 

This would give us a single p-value to assess whether the clustering algorithm is handling sample weight properly or not.

For the sake of simplification, we could do the same for classifiers (for instance, by using the `log_loss` or `brier_score function for models with a `predict_proba` and `roc_auc` for classifiers with `decision_function` only and no `predict_proba`) and `mean_squared_error` for mean regressors and `pinball_loss` for quantile regressors.

For `transformers` that are neither classifiers (e.g. LDA) or clusterers (e.g. KMeans) we could pipeline them with `Ridge` and then compute the distribution of the `mean_squared_error` of the pipeline.

This way, we would have a single p-value for all estimators based on the distribution of their most informative scoring function.

**Comment by ogrisel** [comment]
Rand index and other clustering metrics such as NMI/AMI/ARI/V-measure are invariant to permutation of the cluster labels.

**Comment by snath-xoc** [comment]
Done under #16
